### PR TITLE
Fix broken num_sound_downloads and num_pack_downloads

### PR DESCRIPTION
**Pull request by philtgun**
Adresses #1156 
- Re-implement broken update routines that are bound to django signals
- Add relevant test assertions
- Add recalculation to `report_count_statuses`
- Optimize `downloaded_sounds` and `downloaded_packs` views by passing these fields to paginator to reduce db queries

Would be nice to run updated `report_count_statuses` on test environment just to make sure it is safe to update.

### CODE REVIEW

File: sounds/models.py [hunk 1]
```diff
@@ -1049,7 +1050,7 @@ class Meta:
 
 
 class Download(models.Model):
-    user = models.ForeignKey(User)
+    user = models.ForeignKey(User, related_name='sound_downloads')
```
Comment (ffont) [inline sounds/models.py]: Did you run `makemigrations? I think this changes generate migrations in Django that you should add in the PR.

Comment (philtgun) [reply sounds/models.py]: Good point! Included in next commit

**Comment by ffont** [comment]
Next steps:

- Deploy this code in test env
- Measure how long it takes to run `report_count_statuses` with and without downloads. Compare with the time it takes in production db and report results here.

**Comment by ffont** [comment]
I did run `report_count_statuses` in test env before and after deploying this PR and these are the results.

- Before deploy (run1): 21m, 550 objects updated
- Before deploy (run2): 14m, 0 objects updated
- After deploy (run1): 286m, 275000 objects updated
- After deploy (run2): 221m, 0 objects updated

The query for getting number of downloads per user (https://github.com/MTG/freesound/blob/profile-num-sound-and-pack-downloads/general/management/commands/report_count_statuses.py#L196) takes 3 hours to compute!

I'm not sure how this affects the DB, in test I was able to navigate it while this query was running, but I think it was a bit slower.

The idea is that this command we should just run it from time to time (or ideally never), so maybe it is not a problem if we run it once a week or once every two weeks for example and we get a bit of a slowdown. We can also skip the downloads part of the command (the option is already implemented) and run the other parts once a week and the downloads thing once a month (it's less important). I think we should just deploy to production and run the command and see how Freesound is affected.

**Comment by philtgun** [comment]
Technically after we run this command to fix the number of downloads, if there are no bugs, we don't need to run it again in a while. The cronjob for it is weekly and it skips recalculation of the downloads.

**Comment by ffont** [comment]
Yeah, we just need to run it from time to time so I'm not much worried about it taking 3 hours. But still we should evaluate how much it impacts performance. I'll merge for now and test when deployed to have a better understanding.

**Pull request merged**